### PR TITLE
[Diagnostics] Add a fix-it for misplaced throws in function types

### PR DESCRIPTION
**Pull request by tapthaker**
Adds a diagnostic error for misplaced throws in function types. For e.g:
```swift
let fn: () -> throws Void  // expected-error{{'throws' may only occur before '->'}} {{12-12=throws }} {{15-22=}}
```

Resolves [SR-10703](https://bugs.swift.org/browse/SR-10703).

### CODE REVIEW

**Comment by tapthaker** [comment]
@brentdax: This fixes the bug that you reported. Thanks

File: lib/Parse/ParseType.cpp [hunk 1]
```diff
@@ -417,6 +417,12 @@ ParserResult<TypeRepr> Parser::parseType(Diag<> MessageID,
   if (Tok.is(tok::arrow)) {
     // Handle type-function if we have an arrow.
     SourceLoc arrowLoc = consumeToken();
+    if(Tok.is(tok::kw_throws)) {
```
Comment (jrose-apple) [inline lib/Parse/ParseType.cpp]: Nitpick: Please put a space before the open-paren.

File: lib/Parse/ParseType.cpp [hunk 2]
```diff
@@ -417,6 +417,12 @@ ParserResult<TypeRepr> Parser::parseType(Diag<> MessageID,
...
+     Diag<> DiagID = diag::throws_in_wrong_position;
+     diagnose(Tok.getLoc(), DiagID).fixItInsert(arrowLoc, "throws ")
+	     .fixItRemove(Tok.getLoc());
```
Comment (jrose-apple) [inline lib/Parse/ParseType.cpp]: Looks like a tab snuck in here.

File: lib/Parse/ParseType.cpp [hunk 3]
```diff
@@ -417,6 +417,12 @@ ParserResult<TypeRepr> Parser::parseType(Diag<> MessageID,
...
+     consumeToken();
```
Comment (jrose-apple) [inline lib/Parse/ParseType.cpp]: This is good, but I think we can probably recover as if the user wrote `throws` in the correct place by setting `throwsLoc`.

**Review by jrose-apple — COMMENTED** [review]
Thank you for picking this up! This implementation looks good but I do have a few notes.

**Comment by brentdax** [comment]
@swift-ci please smoke test

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test

**Pull request merged**